### DISCUSSION TITLE
Create a table on external data in SQL

### DISCUSSION THREAD

**Discussion by szst11** [duckdb/duckdb#10762] [Ideas]
Hallo all,
I have a parquet dataset with many partitions and a lot of files. It works well to query the data with `read_parquet` function. This takes a couple of seconds for every task I do. That would be fine for just a few tasks.
But if I use it to query those data with a lot of small tasks, it takes those seconds every time. This is uncomfortable e.g. when using it with [Tableau](https://duckdb.org/docs/guides/data_viewers/tableau). It seems, on every query the full dataset has to be explored again and again.

When I use this dataset in Python, it is possible to import it as a pyarrow dataset. Here every single query is really fast. The exploration of the dataset is done only once.

So, is there a way to create a table on the external data in pure SQL? 
Creating a view is not sufficient, as it is just a view and does not speed up.

**Comment by szarnyasg** [comment]
Hi @szst11, how about using using CTAS (Create Table ... AS)?

```sql
CREATE TABLE <your_table_name> AS FROM read_parquet('<your_parquet_file>');
```

**Reply by szst11** [reply]
Hi @szarnyasg, 
as I understood, this will copy all to the internal database. But the dataset is to large for that.

**Reply by szarnyasg** [reply]
Indeed this will copy the data in the DuckDB database file and also compute some statistics for it that can be later used for query optimization. Indeed this costs extra disk space.

As you have also remarked, it's possible to create views but these will only substitute the `read_parquet` query in subsequent queries.

So I'm afraid at the moment there's no way to create a table-like structure (with statistics) on an external Parquet file.
